### PR TITLE
Reloadsolution

### PR DESCRIPTION
**Pull request by cubitouch**
Adding a "/reloadsolution" endpoint to solve this issue : https://github.com/OmniSharp/omnisharp-roslyn/issues/19

### CODE REVIEW

File: src/OmniSharp/Api/ProjectSystem/OmnisharpController.ProjectSystem.cs [hunk 1]
```diff
@@ -56,5 +60,16 @@ public ProjectInformationResponse CurrentProject(Request request)
                 AspNet5Project = aspNet5ProjectItem
             };
         }
+
+        [HttpPost("/reloadsolution")]
+        [HttpGet("/reloadsolution")]
```
Comment (Yantrio) [inline src/OmniSharp/Api/ProjectSystem/OmnisharpController.ProjectSystem.cs]: Why do we have post and get?

**Comment by filipw** [comment]
there should probably be a restart or reload method on the project system interface. I think it's not a safe assumption to think that `Initialize` will be OK to call over and over

**Comment by Yantrio** [comment]
I agree with @filipw is this open to memory leaks?

**Comment by nosami** [comment]
Doesn't seem to work. 

I asked @cubitouch to submit it so I could test. It was an experiment :)

**Comment by cubitouch** [comment]
So, I will start over following that list :

- Reload "Workspaces" objects
- Reload "Context" objects
- Reload "Projects" collection
- Reload "ProjectMappings" collection

Please let me know if I'm forgetting something, that I can manage that before you review the PR.

**Comment by nosami** [comment]
@cubitouch sounds good. At the _very_ minimum, your code should compile ;) 

After that, I would expect adding new files, projects, deleting files, should just work after firing the reloadsolution command.

File: src/OmniSharp/Api/ProjectSystem/OmnisharpController.ProjectSystem.cs (hunk 1 above)
Comment (davidfowl) [inline src/OmniSharp/Api/ProjectSystem/OmnisharpController.ProjectSystem.cs]: Why do we have a GET

File: src/OmniSharp/AspNet5/AspNet5ProjectSystem.cs [hunk 2]
```diff
@@ -416,7 +416,7 @@ private void TriggerDependeees(string path)
             }
         }
 
-        private void Initialize()
+        private void Startup()
```
Comment (davidfowl) [inline src/OmniSharp/AspNet5/AspNet5ProjectSystem.cs]: What is Startup? That's a bad name.

Comment (cubitouch) [inline src/OmniSharp/AspNet5/AspNet5ProjectSystem.cs]: @davidfowl I don't know more than you... There was a public Initalize() (without the 'i') method on the Interface which I renamed Initialize() (as it should be, IMHO). But there was already a private method with that name so I renamed it too (the Startup one).

What name would you like me to use ? Do you want me to rollback these changes and let the Initalize() method like that ?

**Comment by nosami** [comment]
I've been testing this. It doesn't work reliably yet. Sometimes it works just like you'd expect, but other times it causes intellisense to stop working for the entire solution.

**Comment by cubitouch** [comment]
Is this totally random or some cases just do not work ?
Do you have an idea what can cause the intellisense stop working properly ?

**Comment by nosami** [comment]
I have no idea what is causing the issue because I didn't debug. I spent quite a while testing this though. 

Please test this yourself.

**Comment by troydai** [comment]
@cubitouch this has been stalled for more than a year. Is this PR still valid?

**Comment by cubitouch** [comment]
@troydai no I don't think so, I close it.

**Pull request closed without merging**